### ISSUE TITLE
PANIC: called `Result::unwrap()` on an `Err` value: Any { .. } in query-engine/connectors/sql-query-connector/src/error.rs:58:51

### ISSUE THREAD

**Issue by fkvivid** [prisma/prisma#13800]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.15.0           | 
| OS              | debian-openssl-1.1.x|
| Prisma Client   | 3.9.2              |
| Query Engine    | 0.1.0              |
| Database        | postgresql         |

## Logs
```
47:57.451Z prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client:libraryEngine Searching for Query Engine Library in /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client
prisma:client:libraryEngine loadEngine using /home/market/Desktop/webs/mshop.backend/store-service/node_modules/.prisma/client/libquery_engine-debian-openssl-1.1.x.so.node
prisma:client prisma.queryRaw(
    select 
        reg.store_id,
        store.store_name,
        reg.created_at,
        type.payment_type_status,
        type.payment_type_id
    from sm_store_qpay_reg reg
    inner join sm_store store on reg.store_id = store.store_id and store.status = 'A'
    inner join sm_store_payment_type type on type.payment_type_id = reg.payment_type_id and type.payment_type = 'QPAY' and type.status = 'A'
    where 1=1 and reg.status = 'A' 
    , [])
prisma:client Prisma Client call:
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:client prisma.queryRaw(
    select 
        reg.store_id,
        store.store_name,
        reg.created_at,
        type.payment_type_status,
        type.payment_type_id
    from sm_store_qpay_reg reg
    inner join sm_store store on reg.store_id = store.store_id and store.status = 'A'
    inner join sm_store_payment_type type on type.payment_type_id = reg.payment_type_id and type.payment_type = 'QPAY' and type.status = 'A'
    where 1=1 and reg.status = 'A' 
    order by type.type_status_date desc
    limit $1
    offset $2
    , [10,0])
prisma:client Prisma Client call:
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client prisma.queryRaw(
    select 
        reg.store_id,
        store.store_name,
        reg.created_at,
        type.payment_type_status,
        type.payment_type_id
    from sm_store_qpay_reg reg
    inner join sm_store store on reg.store_id = store.store_id and store.status = 'A'
    inner join sm_store_payment_type type on type.payment_type_id = reg.payment_type_id and type.payment_type = 'QPAY' and type.status = 'A'
    where 1=1 and reg.status = 'A'  and type.payment_type_status = $1
    , [["ACTIVE"]])
prisma:client Prisma Client call:
prisma:client:libraryEngine sending request, this.libraryStarted: true
```

## Client Snippet
```ts
// PLEASE FILL YOUR CODE SNIPPET HERE
```

## Schema
```prisma
// PLEASE ADD YOUR SCHEMA HERE IF POSSIBLE
```

## Prisma Engine Query
```
{"X":{}}
```

**Comment by matthewmueller** [comment]
Hi @fkvivid, can you share the schema and query you were running when you encountered this panic?

**Comment by fkvivid** [comment]
Thanks @matthewmueller  here is a code

```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["interactiveTransactions"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
model sm_store_qpay_reg {
  store_qpay_id         Int      @id(map: "pk_sm_store_qpay_reg") @default(autoincrement())
  store_id              String   @db.VarChar(50)
  payment_type_id       Int
  company_name          String   @db.VarChar(240)
  company_register      String   @db.VarChar(30)
  first_name            String?  @db.VarChar(50)
  last_name             String?  @db.VarChar(50)
  phone_country_code    String?  @default("976") @db.VarChar(30)
  phone                 String?  @db.VarChar(50)
  country_code          String?  @db.VarChar(2)
  level1_id             Decimal? @db.Decimal
  level2_id             Decimal? @db.Decimal
  level3_id             Decimal? @db.Decimal
  level4_id             Decimal? @db.Decimal
  level5_id             Decimal? @db.Decimal
  address1              String?  @db.VarChar(240)
  address2              String?  @db.VarChar(240)
  formatted_address     String?  @db.VarChar(240)
  longitude             String?  @db.VarChar(240)
  latitude              String?  @db.VarChar(240)
  company_purpose       String   @db.VarChar(240)
  business_name         String   @db.VarChar(240)
  platforms             String   @db.VarChar(240)
  con_name              String?  @db.VarChar(50)
  con_phone             String?  @db.VarChar(50)
  banks                 Json?    @db.Json
  pay_partially         Boolean? @default(true)
  min_amount            Decimal? @db.Decimal
  client_id             String?  @db.VarChar(240)
  client_secret         String?  @db.VarChar(240)
  refresh_token         String?  @db.VarChar(240)
  description           String?  @db.VarChar(240)
  status                String   @default("A") @db.VarChar(1)
  object_version_number Decimal  @default(1) @db.Decimal
  created_by            String   @default("anonymous") @db.VarChar(64)
  created_at            DateTime @default(now()) @db.Timestamp(6)
  updated_by            String   @default("anonymous") @db.VarChar(64)
  updated_at            DateTime @default(now()) @db.Timestamp(6)
}
model sm_store {
  store_id                String    @id(map: "pk_sm_store") @default(dbgenerated("uuid_generate_v4()")) @db.Uuid
  store_name              String    @db.VarChar(240)
  company                 String?   @db.VarChar(240)
  company_register        String?   @db.VarChar(20)
  is_company_verified     Boolean   @default(false)
  company_verified_at     DateTime? @db.Date
  industry                String?   @db.VarChar(30)
  my_domain               String    @unique(map: "sm_strore_uk1") @db.VarChar(50)
  use_sub_domain          Boolean?  @default(true)
  domain                  String?   @unique(map: "sm_strore_uk2") @db.VarChar(50)
  is_domain_verified      Boolean?  @default(false)
  domain_date             DateTime? @db.Date
  domain_provider         String?   @db.VarChar(240)
  first_name              String?   @db.VarChar(50)
  last_name               String?   @db.VarChar(50)
  phone_country_code      String?   @default("976") @db.VarChar(30)
  phone                   String?   @db.VarChar(50)
  contact_email           String?   @db.VarChar(240)
  sender_email            String?   @db.VarChar(240)
  store_status            String    @default("ACTIVE") @db.VarChar(30)
  store_status_date       DateTime  @default(now()) @db.Date
  currency_code           String?   @default("MNT") @db.VarChar(30)
  timezone_code           String?   @default("227") @db.VarChar(30)
  unit_system             String?   @default("KG") @db.VarChar(30)
  weight_unit             String?   @db.VarChar(30)
  order_prefix            String?   @db.VarChar(30)
  order_suffix            String?   @db.VarChar(30)
  country_code            String?   @default("MN") @db.VarChar(2)
  level1_id               Int?
  level2_id               Int?
  level3_id               Int?
  level4_id               Int?
  level5_id               Int?
  address1                String?   @db.VarChar(240)
  address2                String?   @db.VarChar(240)
  formatted_address       String?   @db.VarChar(240)
  customer_account        String?   @default("REQUIRED") @db.VarChar(30)
  customer_contact        String?   @default("ONLY_EMAIL") @db.VarChar(30)
  receive_shipping_update String?   @default("ONLY_EMAIL") @db.VarChar(30)
  store_plan_id           String?   @db.VarChar(50)
  img_url                 String?   @db.VarChar(240)
  cart_lock_time          Decimal?  @db.Decimal
  order_lock_time         Decimal?  @db.Decimal
  description             String?   @db.VarChar(240)
  status                  String    @default("A") @db.VarChar(1)
  object_version_number   Decimal   @default(1) @db.Decimal
  created_by              String    @default("anonymous") @db.VarChar(64)
  created_at              DateTime  @default(now()) @db.Timestamp(6)
  updated_by              String    @default("anonymous") @db.VarChar(64)
  updated_at              DateTime  @default(now()) @db.Timestamp(6)
}
model sm_store_payment_type {
  payment_type_id       Int      @id(map: "pk_sm_store_payment_type") @default(autoincrement())
  store_id              String   @db.VarChar(50)
  payment_type          String   @db.VarChar(50)
  payment_type_status   String   @default("PENDING") @db.VarChar(30)
  type_status_date      DateTime @db.Date
  description           String?  @db.VarChar(240)
  status                String   @default("A") @db.VarChar(1)
  object_version_number Decimal  @default(1) @db.Decimal
  created_by            String   @default("anonymous") @db.VarChar(64)
  created_at            DateTime @default(now()) @db.Timestamp(6)
  updated_by            String   @default("anonymous") @db.VarChar(64)
  updated_at            DateTime @default(now()) @db.Timestamp(6)

  @@unique([store_id, payment_type], map: "sm_store_payment_type_uk1")
}



    const data = await db.$queryRaw`
    select 
        reg.store_id,
        store.store_name,
        reg.created_at,
        type.payment_type_status,
        type.payment_type_id
    from sm_store_qpay_reg reg
    inner join sm_store store on reg.store_id = store.store_id and store.status = 'A'
    inner join sm_store_payment_type type on type.payment_type_id = reg.payment_type_id and type.payment_type = 'QPAY' and type.status = 'A'
    where ${condition}
    order by type.type_status_date desc
    limit ${results}
    offset ${page * results}
    `;
```

**Comment by Weakky** [comment]
Hey,

This problem is fixed since 3.14.0. The issue is that you're trying to compare an array (`["ACTIVE"]`) with a column of type `text`.

Versions >= 3.14.0 now print this error:

```
Conversion failed: Couldn't serialize value `Some([Text(Some("ACTIVE"))])` into a `text`. Value is a list but `text` is not.`
```

If you can't upgrade yet, just make sure to fix this part of your query:

```js
and type.payment_type_status = ${"ACTIVE"}
```

instead of

```js
and type.payment_type_status = ${["ACTIVE"]}
```

Thanks for reporting, I'm closing now 🙏
